lib/collection/src/recommendations.rs

```diff
@@ -118,10 +131,17 @@ fn merge_positive_and_negative_avg(
         (VectorInternal::Sparse(positive), VectorInternal::Sparse(negative)) => Ok(positive
             .combine_aggregate(&negative, |pos, neg| pos + pos - neg)
             .into()),
-        (VectorInternal::MultiDense(mut positive), VectorInternal::MultiDense(negative)) => {
-            // merge positive and negative vectors as concatenated vectors with negative vectors negated
-            positive.flattened_vectors.extend(negative.flattened_vectors.into_iter().map(|x| -x));
-            Ok(VectorInternal::MultiDense(positive))
+        (VectorInternal::MultiDense(positive), VectorInternal::MultiDense(negative)) => {
+            // averaged positive and negative multivectors consist of just one vector each
+            debug_assert_eq!(positive.len(), 1);
+            debug_assert_eq!(negative.len(), 1);
+
+            let vector = positive.flattened_vectors.iter().zip(negative.flattened_vectors.iter()).map(|(pos, neg)| pos + pos - neg).collect();
```

Comment:
Just to clarify

If multivectors have different number of vectors, then we take only `len(short_multivector)` vectors from both multivectors and dropping the remainder?